src/features/addTweetActions.ts

```diff
@@ -194,36 +253,119 @@ export const maybeAddTweetActions = makeBTDModule(({settings, TD, jq}) => {
+      .toggleClass('following', following)
+      .find('a[rel="follow"]')
+      .attr('title', TD.ui.template.render('text/follow_action', this))
+      .find('.js-icon-follow')
+      .addClass(addedClass)
+      .removeClass(removedClass);
+  };
+
+  TD.services.TwitterUser.prototype.prettyFollowersCountInTweetAction = function () {
+    return prettyCountInTweetAction(this.followersCount);
+  };
+
+  jq(document).on(
+    'dataFollowStateChange',
+    (
+      _,
+      {
+        action: {userAction},
+        request: {twitterUser},
+      }: {action: {userAction: TwitterActionEnum}; request: {twitterUser: TweetDeckUser}}
+    ) => {
+      if (userAction === 'follow') {
+        twitterUser.setFollowing(true);
+      } else if (userAction === 'unfollow') {
+        twitterUser.setFollowing(false);
+      }
+    }
+  );
+
   jq('body').on(
     'click',
-    '[data-btd-action="mute-account"], [data-btd-action="block-account"]',
+    '[data-btd-action="follow-account"], [data-btd-action="unfollow-account"], [data-btd-action="mute-account"], [data-btd-action="block-account"]',
     (ev) => {
       ev.preventDefault();
       const target = ev.currentTarget;
       if (!isHTMLElement(target)) {
         return;
       }
-      let chirp = getChirpFromElement(TD, ev.target)?.chirp;
+      const chirp = getChirpFromElement(TD, ev.target)?.chirp;
 
       if (!chirp) {
         return;
       }
-      chirp = chirp.targetTweet ? chirp.targetTweet : chirp;
-      const user = chirp.retweetedStatus ? chirp.retweetedStatus.user : chirp.user;
-      const account = chirp.account;
+      const usefulChirp = chirp.retweetedStatus ?? chirp.targetTweet ?? chirp;
+      const user = usefulChirp.retweetedStatus?.user ?? usefulChirp.user;
+      const account = usefulChirp.account;
 
       const action = target.getAttribute('data-btd-action');
 
-      if (action === 'mute-account') {
-        jq(document).trigger('uiMuteAction', {
-          account,
-          twitterUser: user,
-        });
-      } else if (action === 'block-account') {
-        jq(document).trigger('uiBlockAction', {
-          account,
-          twitterUser: user,
+      if (action === 'follow-account' && (settings.showAccountChoiceOnFollow || ev.shiftKey)) {
+        jq(document).trigger('uiShowFollowFromOptions', {
+          userToFollow: user,
         });
+        return;
+      }
+
+      switch (action) {
+        case 'follow-account':
+          jq(document).trigger(usefulChirp.user.following ? 'uiUnfollowAction' : 'uiFollowAction', {
+            account,
+            twitterUser: user,
+          });
+          return;
+
+        case 'mute-account':
+          jq(document).trigger('uiMuteAction', {
+            account,
+            twitterUser: user,
+          });
+          return;
+
+        case 'block-account':
+          jq(document).trigger('uiBlockAction', {
+            account,
+            twitterUser: user,
+          });
+          return;
```

Comment:
Not a blocker: I feel like we'll need to use an `enum` for `action` 🙈

Reply:
I tried Adding an enum `TweetActions` and replacing the actions with its items in 
95c2ea9